phoenix-core/src/it/java/org/apache/phoenix/end2end/ViewUtilIT.java

```diff
@@ -0,0 +1,315 @@
+
+import static org.apache.phoenix.coprocessor.MetaDataProtocol.MIN_SPLITTABLE_SYSTEM_CATALOG;
+import static org.apache.phoenix.jdbc.PhoenixDatabaseMetaData.SYSTEM_LINK_HBASE_TABLE_NAME;
+import static org.apache.phoenix.util.TestUtil.TEST_PROPERTIES;
+import static org.junit.Assert.assertEquals;
+
+public class ViewUtilIT extends ParallelStatsDisabledIT {
+
+    @Test
+    public void testGetSystemTableForChildLinks() throws Exception {
+        assertEquals(SYSTEM_LINK_HBASE_TABLE_NAME, ViewUtil.getSystemTableForChildLinks(
+                MIN_SPLITTABLE_SYSTEM_CATALOG, config));
+
+        // lower version should also give CHILD_LINK table as server upgrade to advanced version
+        assertEquals(SYSTEM_LINK_HBASE_TABLE_NAME, ViewUtil.getSystemTableForChildLinks(
+                MIN_SPLITTABLE_SYSTEM_CATALOG - 1, config));
+    }
+
+    @Test
+    public void testHasChildViewsInGlobalViewCase() throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        TableName catalogOrChildTableName = ViewUtil.getSystemTableForChildLinks(0, config);
+        String schema = generateUniqueName();
+        byte[] schemaInBytes = schema.getBytes(StandardCharsets.UTF_8);
+        byte[] tenantIdInBytes = new byte[0];
+        String fullTableName = schema + "." + generateUniqueName();
+        String secondLevelViewName = schema + "." + generateUniqueName();
+        String thirdLevelViewName4 = schema + "." + generateUniqueName();
+        String leafViewName1 = schema + "." + generateUniqueName();
+        String leafViewName2 = schema + "." + generateUniqueName();
+
+        String tableDDLQuery = "CREATE TABLE " + fullTableName + " (A BIGINT PRIMARY KEY, B BIGINT)";
+        String viewDDLQuery = "CREATE VIEW %s AS SELECT * FROM %s";
+
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.createStatement().execute(tableDDLQuery);
+            conn.createStatement().execute(
+                    String.format(viewDDLQuery, secondLevelViewName, fullTableName));
+            conn.createStatement().execute(
+                    String.format(viewDDLQuery, leafViewName1, secondLevelViewName));
+            conn.createStatement().execute(
+                    String.format(viewDDLQuery, thirdLevelViewName4, secondLevelViewName));
+            conn.createStatement().execute(
+                    String.format(viewDDLQuery, leafViewName2, thirdLevelViewName4));
+
+            try (PhoenixConnection phoenixConnection =
+                    DriverManager.getConnection(getUrl(), props).unwrap(PhoenixConnection.class)) {
+                Table catalogOrChildTable = phoenixConnection.getQueryServices().getTable(
+                        SchemaUtil.getPhysicalName(catalogOrChildTableName.toBytes(),
+                                phoenixConnection.getQueryServices().getProps()).getName());
+
+                assertEquals(true, ViewUtil.hasChildViews(catalogOrChildTable,
```

Comment:
@yanxinyi you can directly use assertTrue or assertFalse instead of assertEquals(true, expression).